column selection topic (#1843)
IgniteUI/igniteui-docfx: en/components/grids_templates/column_selection.md

@@if (igxName === 'IgxGrid') {
---
title: Angular Grid Column Selection | Ignite UI for Angular | infragistics

> _description: Learn how to configure column selection with the Angular Material table. The column selection makes the interaction with the Grid more appealing than ever.

Reviewer: makes grid interactions sounds better IMO

_keywords: column selection, igniteui for angular, infragistics
---
}
@@if (igxName === 'IgxTreeGrid') {
---
title: Angular Tree Grid Column Selection | Ignite UI for Angular | infragistics
_description: Learn how to configure column selection with the Angular Material table. The column selection makes Tree Grid interactions more appealing than ever.
_keywords: column selection, igniteui for angular, infragistics
---
}
@@if (igxName === 'IgxHierarchicalGrid') {
---
title: Angular Hierarchical Grid Column Selection | Ignite UI for Angular | infragistics
_description: Learn how to configure column selection with the Angular Material table. The column selection makes Hierarchical Grid interaction more appealing than ever.
_keywords: column selection, igniteui for angular, infragistics
---
}

> ### @@igComponent Column Selection

Reviewer 2: ```suggestion
## @@igComponent Column Selection
```

> The **column selection feature** provides an easy way to select an entire column with a single click action. It **emphasizes** the importance of a particular column by focusing on the header cell(s) and everything below. The feature comes with a rich [`API`]({environment:angularApiUrl}) that allows manipulation of the selection **state**, **data extraction** from the **selected** fractions and **data analysis** operations and visualizations.

Reviewer 2: ```suggestion
<p class="highlight">The <b>column selection feature</b> provides an easy way to select an entire column with a single click. It <b>emphasizes</b> the importance of a particular column by focusing the header cell(s) and everything below. The feature comes with a rich [`API`]({environment:angularApiUrl}) that allows for manipulation of the selection <b>state</b>, <b>data extraction</b> from the <b>selected</b> fractions and <b>data analysis</b> operations and visualizations.</p>
```

> The **column selection feature** provides an easy way to select an entire column with a one click action. It **emphasizes** the importance of a particular column by focusing on the header cell(s) and everything below. The feature comes with a rich [`API`]({environment:angularApiUrl}) that allows manipulation of the **state**, **data extraction** of the **selected** fractions easing **data analysis** operations and visualizations.

Reviewer: we can consider "with a single click" as well ?

> The **column selection feature** provides an easy way to select an entire column with a one click action. It **emphasizes** the importance of a particular column by focusing on the header cell(s) and everything below. The feature comes with a rich [`API`]({environment:angularApiUrl}) that allows manipulation of the **state**, **data extraction** of the **selected** fractions easing **data analysis** operations and visualizations.

Reviewer: allows manipulation of the selection **state**, **data extraction** from the **selected** fractions and **data analysis** operations and visualizations.

<div class="sample-container loading" style="height:570px">
    <iframe id="grid-column-selection" data-src='{environment:demosBaseUrl}/grid/grid-column-selection' width="100%" height="100%" seamless frameborder="0" class="lazyload no-theming"></iframe>
</div>
<br/>
<div>
<button data-localize="stackblitz" disabled class="stackblitz-btn" data-iframe-id="grid-column-selection-iframe" data-demos-base-url="{environment:demosBaseUrl}">view on stackblitz</button>
</div>
<div class="divider--half"></div>

### Basic usage

#### Interactions

> All of the presented columns are [`selectable`]({environment:angularApiUrl}/classes/igxcolumncomponent.html#selectable) by default. With that being said, in order to select a column, we just need to click on the targeted one, which mark it as [`selected`]({environment:angularApiUrl}/classes/igxcolumncomponent.html#selected).

Reviewer 2: ```suggestion
All of the presented columns are [`selectable`]({environment:angularApiUrl}/classes/igxcolumncomponent.html#selectable) by default. With that being said, in order to select a column, we just need to click on one, which will mark it as [`selected`]({environment:angularApiUrl}/classes/igxcolumncomponent.html#selected).
```

> All of the presented columns are [`selectable`]({environment:angularApiUrl}/classes/igxcolumncomponent.html#selectable) by default. With that being said, in order to select a column, we just need to click over the targeted one to mark it as [`selected`]({environment:angularApiUrl}/classes/igxcolumncomponent.html#selected).

Reviewer: in order to select a column, we just need to click on the targeted on, which mark it as [`selected`]({environment:angularApiUrl}/classes/igxcolumncomponent.html#selected).

> [!NOTE]

> > [`Multi Column Headers`](multi_column_headers.md) don't reflect on the [`selectable`]({environment:angularApiUrl}/classes/igxcolumncomponent.html#selectable) input. The [`IgxColumnGroupComponent`]({environment:angularApiUrl}/classes/igxcolumngroupcomponent.html) is [`selectable`]({environment:angularApiUrl}/classes/igxcolumncomponent.html#selectable), if at least one of it's children has that behaviour enabled. In addition, the component is marked as [`selected`]({environment:angularApiUrl}/classes/igxcolumngroupcomponent.html#selected) if all it's descendents are [`selected`]({environment:angularApiUrl}/classes/igxcolumncomponent.html#selected)

Reviewer 2: ```suggestion
> [`Multi Column Headers`](multi_column_headers.md) don't reflect on the [`selectable`]({environment:angularApiUrl}/classes/igxcolumncomponent.html#selectable) input. The [`IgxColumnGroupComponent`]({environment:angularApiUrl}/classes/igxcolumngroupcomponent.html) is [`selectable`]({environment:angularApiUrl}/classes/igxcolumncomponent.html#selectable), if at least one of its children has the selection behaviour enabled. In addition, the component is marked as [`selected`]({environment:angularApiUrl}/classes/igxcolumngroupcomponent.html#selected) if all of its descendents are [`selected`]({environment:angularApiUrl}/classes/igxcolumncomponent.html#selected)
```

<div class="sample-container loading" style="height:570px">
    <iframe id="grid-column-group-selection" data-src='{environment:demosBaseUrl}/grid/grid-column-group-selection' width="100%" height="100%" seamless frameborder="0" class="lazyload no-theming"></iframe>
</div>
<br/>
<div>
<button data-localize="stackblitz" disabled class="stackblitz-btn" data-iframe-id="grid-column-group-selection-iframe" data-demos-base-url="{environment:demosBaseUrl}">view on stackblitz</button>
</div>
<div class="divider--half"></div>

#### Keyboard combinations

> Basically there are three scenarios for the keyboard navigation of the **Column Selection**:

Reviewer 2: ```suggestion
There are three scenarios for keyboard navigation of the **Column Selection** feature:
```

> - Single selection - <kbd>click</kbd> over the column cell if it is **selectable**.

Reviewer 2: ```suggestion
- Single column selection - <kbd>click</kbd> on the column header cell if it is **selectable**.
```

> - Single selection - <kbd>click</kbd> over the column cell if it is **selectable**.

Reviewer: click on

> - Multi column selection - holding <kbd>ctrl</kbd> + <kbd>click</kbd> over every **selectable** header cell.

Reviewer: click on

- Range column selection - holding <kbd>shift</kbd> + <kbd>click</kbd> selects all **selectable** columns in between.

#### API manipulations

> The **API** provides some additional capabilities when it comes to the **non-visible** columns such that, every **hidden** column could be marked as [`selected`]({environment:angularApiUrl}/classes/igxcolumncomponent.html#selected) by setting the corresponding **setter**.

Reviewer: such as marking every **hidden** column as [`selected`]({environment:angularApiUrl}/classes/igxcolumncomponent.html#selected) by setting the corresponding **setter**.

> [!NOTE]

> > The above statement also applies to the [`IgxColumnGroupComponent`]({environment:angularApiUrl}/classes/igxcolumngroupcomponent.html), except that when the [`selected`]({environment:angularApiUrl}/classes/igxcolumngroupcomponent.html#selected) property is changed it respectively changes the state of it's descendants.

Reviewer 2: ```suggestion
> The above statement also applies to the [`IgxColumnGroupComponent`]({environment:angularApiUrl}/classes/igxcolumngroupcomponent.html), except that when the [`selected`]({environment:angularApiUrl}/classes/igxcolumngroupcomponent.html#selected) property is changed it changes the state of its descendants.
```

More information regarding the API manipulations could be found in the [`API References`](#api-references) section.

### Styling

> Before going deeper in the style manpulations, the core module and all component mixins need to be imported.

Reviewer 2: ```suggestion
Before diving into the styling options, the core module and all component mixins need to be imported.
```

```scss
@import '~igniteui-angular/lib/core/styles/themes/index';
```

>[!NOTE]

> >Please note that [`row selection`](row_selection.md) and [`column selection`](column_selection.md) can't be manipulated   independently. They both are dependent on the same `variables`.

Reviewer 2: ```suggestion
>Please note that [`row selection`](row_selection.md) and [`column selection`](column_selection.md) can't be manipulated   independently. They depend on the same `variables`.
```

> With that being said, let's move on and change the **selection** and **hover** style.<br/>

Reviewer 2: ```suggestion
With that being said, let's move on and change the **selection** and **hover** styles.<br/>
```

> Following the simples approach let's define our custom **theme**.

Reviewer 2: ```suggestion
Following the simplest approach, let's define our custom **theme**.
```

> Following the simples approach let's define our custom **theme**.

Reviewer: the simplest ?

```scss
$custom-grid-theme: igx-grid-theme(
    $row-selected-background: steelblue,
    $row-selected-text-color: gold,
    $row-selected-hover-background: steelblue
);
```

> [`igx-grid-theme`]({environment:sassApiUrl}/index.html#function-igx-grid-theme) accepts several parameters but basically those are the three responsible for the respective styling:

Reviewer 2: ```suggestion
[`igx-grid-theme`]({environment:sassApiUrl}/index.html#function-igx-grid-theme) accepts several parameters but those are the three responsible for changing the appearance of all selected columns:
```

> [`igx-grid-theme`]({environment:sassApiUrl}/index.html#function-igx-grid-theme) accepts several parameters but basically those are the three responsible for the respective styling:

Reviewer: for the selected cells styling ?

- **$row-selected-background** - sets the background of the selected fraction.
- **$row-selected-text-color** - sets the text color of the selected fraction
- **$row-selected-hover-background** - sets the color of the hovered cell or bunch of cells.

#### Using CSS Variables

> The last step is to pass the custom `igx-grid` theme.

Reviewer 2: ```suggestion
The last step is to include the custom `igx-grid` theme.
```

```scss
@include igx-css-vars($custom-grid-theme)
```

> #### Using mixins

Reviewer 2: ```suggestion
#### Overriding the Base Theme
```

> In order to style components for Internet Explorer 11, we have to use different approach, since it doesn't support CSS variables.

Reviewer 2: ```suggestion
In order to style components for Internet Explorer 11, we have to use a different approach, since it doesn't support CSS variables.
```

>[!NOTE]

> >If the component is using an [`Emulated`](../themes/component-themes.md#view-encapsulation) ViewEncapsulation, it is necessary to `penetrate` this encapsulation using `::ng-deep`. In order to prevent the custom theme to leak to other components, be sure that you have included the `:host` selector before `::ng-deep`.

Reviewer 2: ```suggestion
>If the component is using the [`Emulated`](../themes/component-themes.md#view-encapsulation) ViewEncapsulation, it is necessary to `penetrate` this encapsulation using `::ng-deep`. In order to prevent the custom theme from leaking into other components, be sure that you have included the `:host` selector before `::ng-deep`.
```
